File: CCp/TypeScript/src/Math/basic/splitN.ts
function splitN(N: number): void {
  let count = 0;
  while (N > 0) {
    if (N % 2) {
      count++;
    }
    N = Math.floor(N / 2);
  }
  console.log(count - 1);
}

function testSplit() {
  const input = [3, 4, 15];
  //  for( let i = 0; i < input.length; i++ ) {
  //    splitN( input[i]);
  //  }

  for (let i in input) {
    splitN(input[i]);
  }
}

testSplit();

File: CCp/Java/Math/onestar/SplitN.java
package Java.Math.onestar;

import java.util.Scanner;

public class SplitN {

    private static void  printOps(int N ) {
        int count = 0;
        while ( N > 0 ) {
            int rem = N % 2;
            if( rem == 1 ) {
                count++;
            }
            N /= 2;
        }
        System.out.println(count - 1);
    }

    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        int T = s.nextInt();
        for ( int i = 1; i <= T; i++ ) {
            int N = s.nextInt();
            printOps(N);
        }
    }
}
